### PR TITLE
`General`: Handle error when course title too long

### PR DESCRIPTION
**Pull request by sarpsahinalp**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

### Motivation and Context

Currently, there is no limit on the length of course titles. However, when they exceed a certain length, Artemis signals an Internal Server Error, making it unclear what went wrong.

### Description

Give a character limit for the course title in the same way the restrictions for the short name are displayed.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Admin

1. Login as Admin
2. Create new course with very long title > 255 Characters
3. See the error message with title too long
4. Try to update the course with a very long title > 255 Characters
5. See the error message with title too long

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Comment by coolchock** [comment]
I think you should consider the review from coderabbit. When I try to update the course title and set a very long string I also get an Internal Server Error:
image

**Comment by egekurt123** [comment]
"I think we should also handle this on the client side by checking for invalid inputs. Otherwise, it just shows an 'Internal Error' without giving users any information about what went wrong."

**Comment by sarpsahinalp** [comment]
> "I think we should also handle this on the client side by checking for invalid inputs. Otherwise, it just shows an 'Internal Error' without giving users any information about what went wrong."

This is not needed since the error message is also displayed in the UI

**Review by edkaya — APPROVED** [review]
Tested on ts3, works as expected

**Comment by florian-glombik** [comment]
I agree with @egekurt123 and think that it would be better to have a client-side validation as well.

**Review by florian-glombik — CHANGES_REQUESTED** [review]
The translation `titleTooLong` does not seem to exist in the `error.json` on the client side. So the translation will not work, please add the translation there.
I do also think that `courseTitleTooLong` would be a more descriptive translationKey name.

Additionally I think that a client validation should be added. The save option should be invalid if more than 255 chars are entered and the field should have an indication to be invalid.

File: src/main/webapp/app/course/manage/course-update.component.ts [hunk 1]
```diff
@@ -162,7 +162,10 @@ export class CourseUpdateComponent implements OnInit {
         this.courseForm = new FormGroup(
             {
                 id: new FormControl(this.course.id),
-                title: new FormControl(this.course.title, [Validators.required]),
+                title: new FormControl(this.course.title, {
+                    validators: [Validators.required, Validators.maxLength(255)],
```
Comment (florian-glombik) [inline src/main/webapp/app/course/manage/course-update.component.ts:166]: `255` could be encapsulated in a variable as done in the server part

**Review by florian-glombik — COMMENTED** [review]
Thank you for addressing the comments from my previous review!

The formulation of the error message is a matter of taste, so choose the version you like best. 

Since there are no approves on the PR yet, would you mind getting rid of the magic number in the client validation?

**Review by florian-glombik — APPROVED** [review]
Code looks good to me and tested on ts3 - works well 👍 

Thank you for addressing the review comments!

**Review by egekurt123 — APPROVED** [review]
Code LGTM, Manually tested on ts3, works as expected

**Review by az108 — APPROVED** [review]
Tested manually on t3 (branch was still deployed there) , works as expected and Code LGTM

**Review by edkaya — APPROVED** [review]
Reapprove

**Pull request merged**